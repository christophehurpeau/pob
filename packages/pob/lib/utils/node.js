export const latestLTS = "22";
export const maintenanceLTS = "20";
