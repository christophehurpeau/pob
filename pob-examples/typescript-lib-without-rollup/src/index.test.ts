import * as fs from "node:fs";
// import simpleLib from './index.js';

// describe('index', () => {
//   it('should return hello world', () => {
//     expect(simpleLib()).toBe('hello world via typescript');
//   });
// });

describe("dist", () => {
  it("should build index.js", () => {
    expect(
      fs.readFileSync(new URL("../dist/index.js", import.meta.url), "utf8")
      // note that inline snapshot does not work with ESM currently
    ).toMatchSnapshot();
  });
});
