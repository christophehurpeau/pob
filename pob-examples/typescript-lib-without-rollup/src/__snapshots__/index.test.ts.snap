// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dist should build index.js 1`] = `
"export default function simpleLib() {
    // eslint-disable-next-line @typescript-eslint/no-unnecessary-template-expression -- checking template string here
    return \`hello world via \${"typescript"}\`;
}
//# sourceMappingURL=index.js.map"
`;
